core/src/main/java/com/datastax/oss/driver/internal/core/config/map/MapBasedDriverConfigLoader.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.config.map;
+
+import com.datastax.dse.driver.api.core.config.DseDriverOption;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.config.DriverOption;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MapBasedDriverConfigLoader implements DriverConfigLoader {
+
+  private final Map<String, Map<DriverOption, Object>> optionsMap;
+
+  public MapBasedDriverConfigLoader(@NonNull Map<String, Map<DriverOption, Object>> optionsMap) {
+    this.optionsMap = optionsMap;
+  }
+
+  @NonNull
+  @Override
+  public DriverConfig getInitialConfig() {
+    return new MapBasedDriverConfig(optionsMap);
+  }
+
+  @Override
+  public void onDriverInit(@NonNull DriverContext context) {
+    // nothing to do
+  }
+
+  @NonNull
+  @Override
+  public CompletionStage<Boolean> reload() {
+    return CompletableFuture.completedFuture(true);
+  }
+
+  @Override
+  public boolean supportsReloading() {
+    return true;
+  }
+
+  @Override
+  public void close() {
+    // nothing to do
+  }
+
+  public static Map<String, Map<DriverOption, Object>> buildDefaultOptionsMap() {
+    Map<String, Map<DriverOption, Object>> optionsMap = new ConcurrentHashMap<>();
+    Map<DriverOption, Object> defaultProfileMap = new ConcurrentHashMap<>();
+    optionsMap.put(DriverExecutionProfile.DEFAULT_NAME, defaultProfileMap);
+
+    // Skip CONFIG_RELOAD_INTERVAL because the map-based config doesn't need periodic reloading
```

Comment:
I'd like to avoid a dependency to Typesafe config at runtime, i.e. the user can exclude it in their POM if they use the map-based loader.

The unit test I commented on above ensures that we don't forget to keep the two places in sync: the build will fail if both sides don't match.

In fact I also tested that when I forgot to add the new `LOAD_BALANCING_POLICY_SLOW_AVOIDANCE` option when rebasing my PR.

Reply:
nice idea with that unit test